fix(format): stop adding an extra line ending

GmoZen opened the pull request:
Fixes https://github.com/prisma/prisma/issues/8548.

@DanielZavala11 and I propose removing os.EOL (line ending specific to the operating system) from the end of the Format.ts file's output, and replacing it with a "\n" line ending. This ensures that the output has consistent line endings for every line. The current behavior is that the final line in the file will have an OS dependent line ending, but all other lines will have "\n".

Our reasoning is that this leads to more consistency, and matches the last line ending with other line endings in the output, which are already being normalized to "/n" in the prisma-engines repo (line 69, [reformat.rs](http://reformat.rs/) file):

https://github.com/prisma/prisma-engines/blob/b4538d597e8760ef8b6c235ed3127f82be2db0fb/libs/datamodel/core/src/reformat.rs#L69

GmoZen committed:
Standardized on Unix line endings

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prisma/prisma?pullRequest=12401) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prisma/prisma?pullRequest=12401) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prisma/prisma?pullRequest=12401) it.</sub>

Jolg42 commented on packages/cli/src/Format.ts:
```diff
@@ -76,7 +75,7 @@ Or specify a Prisma schema path
     })
 
-    output = output.trimEnd() + os.EOL
+    output = output.trimEnd() + '\n'
 
```
```suggestion
```

From @tomhoule 

Since it's already done by https://github.com/prisma/prisma-engines/blob/15c0579cf8f153388368beadf2185058b7135a17/libs/datamodel/core/src/reformat.rs#L71

We could remove the line instead

Jolg42 replied:
And then `let output` can be turned into `const output`

Jolg42 reviewed (changes requested):
Thanks for the PR!

Check the suggested change before we can merge 👍🏼

GmoZen committed:
Update packages/cli/src/Format.ts

Remove line that adds line feed character to end of output

Co-authored-by: Joël Galeran <Jolg42@users.noreply.github.com>

GmoZen committed:
Switch let with const for output variable

janpio commented:
This is now making 2 tests fail which need to either be changed or removed:
```
[2022-03-22T21:34:17Z]  FAIL  src/__tests__/commands/Format.test.ts
[2022-03-22T21:34:17Z]   ✕ format should add a trailing EOL (377 ms)
[2022-03-22T21:34:17Z]   ✕ format should add missing backrelation (316 ms)
```

DanielZavala11 committed:
fix: fix the path to the schema missing-backrelation.prisma

This fixes the path by adding temporary directory to the path so that
the function getSchemaPathInternal can return a viable path to the schema
to be formatted for testing.

BREAKING CHANGE
Before this change the 'format should add missing backrelation' failed
automatically, behavior changes from the error --schema at
missing-backrelation.prisma doesn't exist to Prisma schema loaded from
schema.prisma to be compared to the snapshot.

DanielZavala11 committed:
Revert "fix: fix the path to the schema missing-backrelation.prisma"

Change commit from Formatting branch to main branch so that the change is
in the same pull request that has Format.ts change.

This reverts commit bc81da4600ccc72deca7c97a1443a905286a31c8.

DanielZavala11 committed:
fix: fix the path to the schema missing-backrelation.prisma
This fixes the path by adding temporary directory to the path so that
the function getSchemaPathInternal can return a viable path to the schema
to be formatted for testing.

BREAKING CHANGE
Before this change the 'format should add missing backrelation' failed
automatically, behavior changes from the error --schema at
missing-backrelation.prisma doesn't exist to Prisma schema loaded from
schema.prisma to be compared to the snapshot.

DanielZavala11 committed:
Merge branch 'main' of https://github.com/codeday-init/prisma into main
to merge the remote branch into mine

janpio commented:
<img width="334" alt="image" src="https://user-images.githubusercontent.com/183673/162467504-be5893d6-7341-4be1-9093-3d1c1309f2e5.png">
on our side, so can not retrigger the GH Actions tests unfortunately (But Buildkite is still failing)

DanielZavala11 committed:
deleted the last line feeds from 'format should add missing backrelation' and 'format should add a trailing EOL 1'
changed title from `format should add a trailing EOL 1` to `format should have consistent EOL for all lines`
changes reflect the removal of output's trailing EOL from Format.ts

janpio commented on packages/cli/src/__tests__/commands/__snapshots__/Format.test.ts.snap:
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`format should add a trailing EOL 1`] = `
+exports[`format should have consistent EOL for all lines`] = `
```
```suggestion
exports[`format should have consistent EOL for all lines 1`] = `
```
The one is automatically added for the first snapshot here I think and is needed.

DanielZavala11 committed:
Update packages/cli/src/__tests__/commands/__snapshots__/Format.test.ts.snap

Co-authored-by: Jan Piotrowski <piotrowski+github@gmail.com>

DanielZavala11 committed:
resolve branch conflict

DanielZavala11 committed:
Merge branch 'main' into main

wmadden commented:
Hi @GmoZen,

Thank you for your contribution and for addressing the issue of the inconsistent line ending in the formatter. We apologize for the delay in reviewing your pull request. The changes you’ve implemented look fine and all tests passed last time they run. 

Could you please update your branch with the latest changes from main to resolve any potential conflicts? Once that’s done, we'd be happy to merge your PR.

We appreciate your patience and effort in improving Prisma!

aqrln commented:
Ping @GmoZen. We'd like to merge it but we can't update the branch ourselves to make it possible because the pull request was opened from the `main` branch of your fork. Going forward, please make sure to create separate branches for your PRs to contribute to open source projects on GitHub, PRs opened from `main` cause troubles for maintainers.

aqrln commented:
Looking at why exactly it conflicts, an equivalent change was already (accidentally?) done a few months after this PR in https://github.com/prisma/prisma/pull/14766. I'll go ahead and close this. Thank you again for the pull request and apologies it wasn't merged timely.
